internal/core/thirdparty/tantivy/tantivy-binding/src/index_reader_text_c.rs

```diff
@@ -18,23 +24,24 @@ pub extern "C" fn tantivy_match_query(ptr: *mut c_void, query: *const c_char) ->
 pub extern "C" fn tantivy_register_tokenizer(
     ptr: *mut c_void,
     tokenizer_name: *const c_char,
-    tokenizer_params: *mut c_void,
+    tokenizer_params: *const c_char,
 ) {
+    init_log();
     let real = ptr as *mut IndexReaderWrapper;
     let tokenizer_name_str = unsafe { CStr::from_ptr(tokenizer_name) };
     let analyzer = unsafe {
-        let m = tokenizer_params as *const HashMap<String, String>;
-        create_tokenizer(&(*m))
+        let params = c_str_to_str(tokenizer_params).to_string();
+        create_tokenizer(&params)
     };
     match analyzer {
-        Some(text_analyzer) => unsafe {
+        Ok(text_analyzer) => unsafe {
             (*real).register_tokenizer(
                 String::from(tokenizer_name_str.to_str().unwrap()),
                 text_analyzer,
             );
         },
-        None => {
-            panic!("unsupported tokenizer");
-        }
+        Err(err) => {
+            panic!("create tokenizer failed with error: {}", err.to_string());
```

Comment:
does golang catch this panic and return to user?

Reply:
No, now all rust error don't return to user, we need a rust status return to c like cstatus return to go, to fetch error.